### DISCUSSION TITLE
The beginnnings of an idea for a variation?

### DISCUSSION THREAD

**Discussion by windowsrefund** [ramojus/mellifluous.nvim#69] [General]
Greets,

I have almost no idea what I'm doing but think I'm onto something. Before going too crazy, I figured I'd share thoughts and ask for some help.

So I'm not especially fond of how things look as far as the floating windows are concerned. The background just feels too bright which in turn, makes the foreground text more difficult to read than need be when they're set against the even lighter foreground of a selected entry. For example, we see this with completion:

![image](https://github.com/user-attachments/assets/a0e28500-e590-4ee4-bb07-e436adfc555e)

Granted, I totally understand how subjective this is but for me (several feet away from my monitor), it's not exactly easy to distinguish the `aa` compared to the `oo` on the next line. Issues involving the aging eyes aside, I'm simply talking about the bright-on-bright aspect.... OK so then I also am not especially fond of how a `:GitSIgns preview_hunk` is rendered:

![image](https://github.com/user-attachments/assets/517948ed-df51-48fd-8fc0-6f3c5b134838)

It seems to me the grey background color serving more of a functional role in terms of distinguishing the area of the thing. I just don't like it because (as shown) it blends with the exact same color of CursorLine which kinda "blends" things in a way that.... irks. OK so I figured I'd see how these things look in another established them like Astronvim's default....

![image](https://github.com/user-attachments/assets/383b8ec2-f748-4f5d-9e6c-b96391d6b178)

Aha! Much better. the entire area is darkened making it impossible to clobber against CursorLine and we have a nice border to help with definition.

![image](https://github.com/user-attachments/assets/d79756d7-f8bb-485b-a6c9-d7ed2c5fa1ab)

Yep, looks good

So I started to explore adding some `highlight_overrides`. I figured out `FloatBorder` and have been using trial & error to figure out the Cmp groups (I'm obviously not entirely there yet). Basically, I've got enough to see a light at the end of the tunnel but I am changing `bg` colors which I don't believe is the typical way of things as far as this theme is concerned.

```
      highlight_overrides = {
        dark = function(hl, colors)
          hl.set("StatusLine", { bg = colors.dark_bg })
          hl.set("Directory", { fg = colors.ui_blue })
          hl.set("FloatBorder", { fg = colors.fg5 })
          hl.set("CmpItemAbbr", { bg = colors.bg })
          hl.set("CmpItemKind", { bg = colors.bg })
          hl.set("CmpItemMenu", { bg = colors.bg })
          -- hl.set("CmpItemAbbrMatch", { fg = hl.get("Pmenu").fg })
          -- hl.set("CmpItemAbbrMatchFuzzy", { link = "CmpItemAbbrMatch" })
          for _, hl_group in pairs(require("oil-git-status").highlight_groups) do
            hl.set(hl_group.hl_group, { bg = colors.bg })
          end
        end,
      },
```

![image](https://github.com/user-attachments/assets/4a859585-02f1-4c2e-b2fe-1434320813a4)

I absolutely see the potential there if I could just replace the bright gray of the first matched line with the gray of the 2nd line and then remove the gray highlight from the 2nd as well. As shown by the commented config, I still need to get the match going and there are other groups to identify...

![image](https://github.com/user-attachments/assets/bef8caba-fc74-4c38-b463-2eb75255593c)

Again, feels like it's heading in the right direction as long as I remove that gray background of `FloatTitle`.

Happy to keep digging around on my end but am very interested in hearing thoughts as I tend to think this style could justify a new standard variation?

**Comment by windowsrefund** [comment]
Aha! Pretty happy with this

```
      highlight_overrides = {
        dark = function(hl, colors)
          hl.set("Pmenu", { link = "LineNr" })
          hl.set("PmenuSel", { link = "CursorLine" })
          hl.set("NormalFloat", { link = "LineNr" })
          hl.set("FloatBorder", { link = "LineNr" })
        end,
      },
```

![image](https://github.com/user-attachments/assets/83a17637-a08c-4548-8982-a02e892c4d8b)

![image](https://github.com/user-attachments/assets/f2f99a5f-568b-4fda-8cc1-04311e2ac380)

![image](https://github.com/user-attachments/assets/d24d0348-6073-43e6-9c94-c74af29e6c9c)

**Comment by ramojus** [comment]
Some pretty good observations here, thanks!

Yesterday I made some improvements to Gitsigns [here](https://github.com/ramojus/mellifluous.nvim/pull/70). Some highlight groups for this plugin were simply not considered before, so I added them now. I think it looks much better now.

Selected item in completion menu does seem to have a background that's a bit too bright. I agree that it's hard to read the text in selected item. I'll take a look at this as well.

As for the general concern of floating windows being too bright, I like the minimalism of it (there is no need of borders then). If you want non-bright floating windows with borders, there is a config for that, I call it flat background:
```lua
require('mellifluous').setup({
    flat_background = { floating_windows = true },
})
```
However, it will not make completion menu flat... I think I did it this way because the completion menu does not have borders by default.
There will also be a problem with `Lazy` and `Mason`, as they also don't have borders by default, but they are made flat by this setting. They both have the same setting for this though:
```lua
ui = {
    -- The border to use for the UI window. Accepts same border values as |nvim_open_win()|.
    border = "none", -- e.g. try "single"
}
```

I think `flat_background` setting should be configurable for all of these separately (`Pmenu`, `Lazy` and `Mason`). I'll work on that when I have time, but if you want, feel free to make a PR.

TLDR: the variation exists and it's configured with `flat_background` setting, but there are some things that need improvement.

**Comment by ramojus** [comment]
Just to clarify, the border for `Mason` and `Lazy` is configured in their own configs, Mellifluous itself does not have a setting for that, only a setting for flat background

**Comment by ramojus** [comment]
Completion menu readability was improved with https://github.com/ramojus/mellifluous.nvim/pull/73. I hope you agree that both the completion menu and gitsigns preview look better now. These are things that I myself got used to and unconsciously learned to ignore, so I'm glad that you voiced your opinion.

**Reply by windowsrefund** [reply]
Thank you so much for the help and improvments :+1:
